File: python/TwoSums.py
class TwoSums(object):
    def twoSum(self, nums, target):
        twoSumMap = {}

        for i in range(len(nums)):
            complement = target - nums[i]

            if complement in twoSumMap:
                return [twoSumMap[complement], i]
            
            twoSumMap[nums[i]] = i

        return []
     
print(TwoSums().twoSum([2, 7, 11, 15], 9))

File: c++/TwoSums.cpp
#include <iostream>
#include <vector>
#include <map>

using namespace std;

class TwoSums {
public:
    vector<int> twoSum (vector<int>& nums, int target) {
        map<int, int> twoSumMap;

        for (int i = 0; i < nums.size(); i++) {
            int complement = target - nums[i]; 

            if (twoSumMap.find(complement) != twoSumMap.end()) {
                return {twoSumMap[complement], i};  
            }

            twoSumMap[nums[i]] = i;  
        }
       
        return {};
    }
};
 
int main() {
    TwoSums twosums;
    vector<int> nums = {2, 7, 11, 15};
    int target = 9;
    vector<int> result = twosums.twoSum(nums, target);
    
    cout << "Indices: [" << result[0] << ", " << result[1] << "]" << endl;
    
    return 0;
}


